cloudflare/src/framework/blocking_api.rs

```diff
@@ -14,10 +15,19 @@ impl HttpApiClient {
         config: HttpApiClientConfig,
         environment: Environment,
     ) -> anyhow::Result<HttpApiClient> {
-        let http_client = reqwest::blocking::Client::builder()
+        let mut builder = reqwest::blocking::Client::builder()
             .timeout(config.http_timeout)
-            .default_headers(config.default_headers)
-            .build()?;
+            .default_headers(config.default_headers);
+
+        if let Some(address) = config.resolve_ip {
+            let url = url::Url::from(&environment);
+            builder = builder.resolve(
+                url.host_str()
+                    .expect("Environment url should have a hostname"),
+                SocketAddr::new(address, 443),
```

Comment:
I think the port (443) should be configurable too, what if user want to resolve the given URL with a different port?

Reply:
I added it for clarity, but it's actually ignored by reqwest.
https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.resolve

Reply:
I see, this is _not_ specifying a URL for a DNS resolver, rather, it's letting users skip DNS resolution and instead manually supply a hostname -> IP mapping themselves.